### PR TITLE
Favorite Screen

### PR DESCRIPTION
**Pull request by anhnv-0990**
## Related Tickets

- [#23493](https://edu-redmine.sun-asterisk.vn/issues/23493)

## What's this PR do ?

- [ ] Create UI for Favorite Screen: 1 TableView
- [ ] Handle Event for Favorite Screen: Delete and Select Movie
- [ ] Display Data for Favorite Screen: Get all movies from Realm

## Library
- RealmSwift
- RxRealm

### CODE REVIEW

File: MFlix/MFlix/Scenes/Favorite/FavoriteViewController.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+//  MFlix
+//
+//  Created by Viet Anh on 5/6/20.
+//  Copyright © 2020 VietAnh. All rights reserved.
+//
+
+final class FavoriteViewController: UIViewController, BindableType {
+    
+    @IBOutlet private weak var tableView: UITableView!
+    var viewModel: FavoriteViewModel!
```
Comment (longvt-1468) [inline MFlix/MFlix/Scenes/Favorite/FavoriteViewController.swift:15]: break line

File: MFlix/MFlix/Scenes/Favorite/FavoriteViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        let isEmptyMovie: Driver<Bool>
+        var deletedMovie: Driver<Bool>
+    }
+    
+    func transform(_ input: Input) -> Output {
+        
+        let movies = input.loadTrigger
+            .flatMapLatest { _ in
+                self.useCase.getAllMovies()
+                .asDriverOnErrorJustComplete()
```
Comment (longvt-1468) [inline MFlix/MFlix/Scenes/Favorite/FavoriteViewModel.swift]: ident

File: MFlix/MFlix/Scenes/Favorite/FavoriteViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            }
+        
+        let selectedMovie = input.selectedMovieTrigger
+            .withLatestFrom(movies) { $1[$0.row] }
+            .do(onNext: {
+                self.navigator.toMovieDetailScreen(movie: $0)
+            })
+        
+        let isEmptyMovie = movies
+            .map{ $0.isEmpty }
```
Comment (longvt-1468) [inline MFlix/MFlix/Scenes/Favorite/FavoriteViewModel.swift]:
```suggestion
            .map { $0.isEmpty }
```

File: MFlix/MFlix/Scenes/Favorite/FavoriteViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        
+        let deletedMovie = input.deletedMovieTrigger
+            .withLatestFrom(movies) { $1[$0.row] }
+            .flatMapLatest {
+                self.navigator.showAlertDelete(movie: $0)
+                    .asDriverOnErrorJustComplete()
+            }
+            .flatMapLatest {
+                self.useCase.delete($0)
+                .asDriverOnErrorJustComplete()
```
Comment (longvt-1468) [inline MFlix/MFlix/Scenes/Favorite/FavoriteViewModel.swift]: ident

**Review by dongtg-1485 — APPROVED** [review]

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**